Sources/TuistGenerator/Generator/SchemeDescriptorsGenerator.swift

```diff
@@ -439,8 +437,17 @@ final class SchemeDescriptorsGenerator: SchemeDescriptorsGenerating {
 
         let buildConfiguration = scheme.runAction?.configurationName ?? defaultBuildConfiguration
         let disableMainThreadChecker = scheme.runAction?.diagnosticsOptions.contains(.mainThreadChecker) == false
+
+        let launchActionConstants: Constants.LaunchAction
+        let debuggerIdentifier: String
         let isSchemeForAppExtension = self.isSchemeForAppExtension(scheme: scheme, graphTraverser: graphTraverser)
-        let launchActionConstants: Constants.LaunchAction = isSchemeForAppExtension == true ? .extension : .default
+        if isSchemeForAppExtension == true {
+            launchActionConstants = .extension
+            debuggerIdentifier = ""
+        } else {
+            launchActionConstants = .default
+            debuggerIdentifier = scheme.runAction?.attachDebugger == true ? XCScheme.defaultDebugger : ""
```

Comment:
This has a subtle issue - by default many schemes may omit explicitly specifying a `runAction` (i.e. it will be `nil`) - as such we'd be disabling the debugger option for them whereas before it was enabled :/ 

An example of this is the automatically generated `-Project` workspace scheme.

<img width="1512" alt="project-scheme" src="https://user-images.githubusercontent.com/11914919/145561622-eef529c6-6ec6-4839-aeea-22ce513ee67d.png">


```suggestion
            if let runAction = scheme.runAction {
                debuggerIdentifier = runAction.attachDebugger ? XCScheme.defaultDebugger : ""
            } else {
                debuggerIdentifier = XCScheme.defaultDebugger
            }
```

It may be worth expanding the tests to cover this case if they don't already.

Reply:
Good find 😲  I fixed this as suggested and add a test for this 👍